api/v1alpha1/storageclassclaim_types.go

```diff
@@ -20,6 +20,22 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// StorageClassClaimState represent a StorageClassClaimState's state
+type StorageClassClaimState string
+
+const (
+	// StorageClassClaimStateReady represents Ready state of StorageClassClaimState
+	StorageClassClaimStateReady StorageClassClaimState = "Ready"
+	// StorageClassClaimStateConfiguring represents Configuring state of StorageClassClaimState
+	StorageClassClaimStateConfiguring StorageClassClaimState = "Configuring"
+	// StorageClassClaimStateDeleting represents Deleting state of StorageClassClaimState
+	StorageClassClaimStateDeleting StorageClassClaimState = "Deleting"
+	// StorageClassClaimStateFailed represents Failed state of StorageClassClaimState
+	StorageClassClaimStateFailed StorageClassClaimState = "Failed"
+	// StorageClassClaimStateDisabled represents Disabled state of StorageClassClaimState
+	StorageClassClaimStateDisabled StorageClassClaimState = "Disabled"
```

Comment:
What is a `Failed` and `Disabled` state, how do we get there?

Reply:
we are mark `Disabled` when `! storageConsumer.spec.Enable` and probably I can remove the `Failed` I kept following storageconsumer but that also doesn't seem to be using it. So, I'll remove the `Failed`